src/Akeneo/Bundle/ElasticsearchBundle/Cursor/FromBoundedCursor.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Akeneo\Bundle\ElasticsearchBundle\Cursor;
+
+use Akeneo\Bundle\ElasticsearchBundle\Client;
+use Akeneo\Component\StorageUtils\Cursor\CursorInterface;
+use Akeneo\Component\StorageUtils\Repository\CursorableRepositoryInterface;
+
+/**
+ * Bounded cursor to iterate over items where a start and a limit are defined.
+ * Internally, this is implemented with the from/size pagination.
+ * {@see https://www.elastic.co/guide/en/elasticsearch/reference/5.x/search-request-from-size.html}
+ *
+ * @author    Julien Janvier <jjanvier@akeneo.com>
+ * @author    Marie Bochu <marie.bochu@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class FromBoundedCursor extends AbstractCursor implements CursorInterface
+{
+    /** @var int */
+    protected $from;
+
+    /** @var int */
+    protected $limit;
+
+    /** @var int */
+    protected $to;
+
+    /** @var int */
+    protected $fetchedItemsCount;
+
+    /**
+     * @param Client                        $esClient
+     * @param CursorableRepositoryInterface $repository
+     * @param array                         $esQuery
+     * @param string                        $indexType
+     * @param int                           $pageSize
+     * @param int                           $limit
+     * @param int                           $from
+     */
+    public function __construct(
+        Client $esClient,
+        CursorableRepositoryInterface $repository,
+        array $esQuery,
+        $indexType,
+        $pageSize,
+        $limit,
+        $from = 0
+    ) {
+        $this->esClient = $esClient;
+        $this->repository = $repository;
+        $this->esQuery = $esQuery;
+        $this->indexType = $indexType;
+        $this->pageSize = $pageSize;
+        $this->limit = $limit;
+        $this->from = $from;
+        $this->to = $this->from + $this->limit;
+
+        $this->items = $this->getNextItems($esQuery);
```

Comment:
actually we decided with @skeleton to revert my fix
it made the code really less readable

I had to do something like
```php
        if (null === $this->items) {
            $this->items = $this->getNextItems($this->esQuery);
        }
```

in each public method of the iterator (as you don't know which one will be called first)

as those cursors classes are NOT services, I think it's acceptable to keep this call in the constructor

otherwise, I would have agreed with you